posthog/api/hog_function.py

```diff
@@ -135,7 +135,7 @@ def validate(self, attrs):
                     {"template_id": "The Data Pipelines addon is required to create custom functions."}
                 )
 
-            if template.status != "free":
+            if template.status != "free" and self.context.get("view") and self.context["view"].action == "create":
```

Comment:
We can just use the instance reference to determine if this is an update or not

```suggestion
            if template.status != "free" and not instance:
```